sdk/resourcemanager/azure-resourcemanager-trafficmanager/src/main/java/com/azure/resourcemanager/trafficmanager/fluent/models/EndpointInner.java

```diff
@@ -5,123 +5,35 @@
-     * endpoint health and is included in the traffic routing method.
-     */
-    @JsonProperty(value = "properties.endpointStatus")
-    private EndpointStatus endpointStatus;
-
-    /*
-     * The weight of this endpoint when using the 'Weighted' traffic routing
-     * method. Possible values are from 1 to 1000.
-     */
-    @JsonProperty(value = "properties.weight")
-    private Long weight;
-
-    /*
-     * The priority of this endpoint when using the 'Priority' traffic routing
-     * method. Possible values are from 1 to 1000, lower values represent
-     * higher priority. This is an optional parameter.  If specified, it must
-     * be specified on all endpoints, and no two endpoints can share the same
-     * priority value.
-     */
-    @JsonProperty(value = "properties.priority")
-    private Long priority;
-
-    /*
-     * Specifies the location of the external or nested endpoints when using
-     * the 'Performance' traffic routing method.
-     */
-    @JsonProperty(value = "properties.endpointLocation")
-    private String endpointLocation;
-
-    /*
-     * The monitoring status of the endpoint.
-     */
-    @JsonProperty(value = "properties.endpointMonitorStatus")
-    private EndpointMonitorStatus endpointMonitorStatus;
-
+public final class EndpointInner extends ProxyResource {
     /*
-     * The minimum number of endpoints that must be available in the child
-     * profile in order for the parent profile to be considered available. Only
-     * applicable to endpoint of type 'NestedEndpoints'.
+     * The properties of the Traffic Manager endpoint.
      */
-    @JsonProperty(value = "properties.minChildEndpoints")
-    private Long minChildEndpoints;
+    @JsonProperty(value = "properties")
+    private EndpointProperties innerProperties;
 
-    /*
-     * The list of countries/regions mapped to this endpoint when using the
-     * 'Geographic' traffic routing method. Please consult Traffic Manager
-     * Geographic documentation for a full list of accepted values.
-     */
-    @JsonProperty(value = "properties.geoMapping")
-    private List<String> geoMapping;
-
-    /*
-     * The list of subnets, IP addresses, and/or address ranges mapped to this
-     * endpoint when using the 'Subnet' traffic routing method. An empty list
-     * will match all ranges not covered by other endpoints.
-     */
-    @JsonProperty(value = "properties.subnets")
-    private List<EndpointPropertiesSubnetsItem> subnets;
-
-    /*
-     * List of custom headers.
-     */
-    @JsonProperty(value = "properties.customHeaders")
-    private List<EndpointPropertiesCustomHeadersItem> customHeaders;
-
-    /*
-     * Fully qualified resource Id for the resource. Ex -
-     * /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/trafficManagerProfiles/{resourceName}
-     */
-    @JsonProperty(value = "id")
-    private String id;
```

Comment:
Removal of `id`, `name` and `type` seems not correct. 

Did you merge the latest autorest.java main? I had an PR merged earlier to address this issue. If so, I have to take another look...

Reply:
OK, I saw the fix, never mind.

Reply:
Yeah, the PR wasn't merged with the ProxyResource fix.